### PR TITLE
20 Creating DSL from service flow builder

### PR DESCRIPTION
**Pull request by Minwasko**

### CODE REVIEW

File: DSL/Node/src/conversion.ts [hunk 1]
```diff
@@ -5,21 +5,31 @@ import Papa from "papaparse";
...
+router.post("/json_to_yaml", (req: Request, res: Response) => {
+  const result = stringify(req.body);
+  res.send({ json: result });
 });
 
-router.post('/json_to_yaml', (req: Request, res: Response) => {
-    const result = stringify(req.body);
-    res.send({"json": result});
+router.post("/json_to_yaml_data", (req: Request, res: Response) => {
+  console.log(req.body.data);
```
Comment (1AhmedYasser) [inline DSL/Node/src/conversion.ts:30]: Please remove console.log

File: DSL/Node/src/conversion.ts [hunk 2]
```diff
@@ -5,21 +5,31 @@ import Papa from "papaparse";
...
-    res.send(result.data);
+router.post("/csv_to_json", multer().array("file"), (req, res) => {
+  const file = base64ToText(req.body.file);
+  const result = Papa.parse(file, { skipEmptyLines: true });
+  res.send(result.data);
 });
 
-router.post('/yaml_to_json', multer().array('file'), (req: Request, res: Response) => {
+router.post(
+  "/yaml_to_json",
```
Comment (1AhmedYasser) [inline DSL/Node/src/conversion.ts:15]: Try using format on save to make the code more formatted as csv_to_json is formatted differently that yam_to_json

**Review by 1AhmedYasser — CHANGES_REQUESTED** [review]

File: DSL/Node/src/conversion.ts (hunk 2 above)
Comment (Minwasko) [reply DSL/Node/src/conversion.ts:15]: It is already a result of formatting, this line is longer 😅

File: DSL/Ruuter/POST/saveJsonToYml.yml
Comment (trevorling) [inline DSL/Ruuter/POST/saveJsonToYml.yml:1]: Is this file actually needed?

File: GUI/src/resources/api-constants.ts [hunk 3]
```diff
@@ -58,3 +58,5 @@ export const servicesRequestsExplain = (): string => `${baseUrl}/services/reques
 export const getClientInputAvailableVariables = (): string => `${baseUrl}/mocks/client-input-variables`;
 export const getTaraAuthResponseVariables = (): string => `${baseUrl}/mocks/tim/user-info`;
 export const getEndpointValidationMock = (): string => `${baseUrl}/mocks/validation-mock`;
+export const jsonToYml = (): string => `${baseUrl}/saveJsonToYml`;
```
Comment (trevorling) [inline GUI/src/resources/api-constants.ts:61]: Is it needed?

**Review by trevorling — APPROVED** [review]

File: GUI/src/resources/api-constants.ts (hunk 3 above)
Comment (Minwasko) [reply GUI/src/resources/api-constants.ts:61]: This is currently to save endpoints and their info/config. Ideally this logic could be a part of services/add but let's keep it simple until Rainer gives us a better overview of what a finished solution should look like.

File: DSL/Ruuter/POST/saveJsonToYml.yml
Comment (Minwasko) [reply DSL/Ruuter/POST/saveJsonToYml.yml:1]: This is currently to save endpoints and their info/config. Ideally this logic could be a part of services/add but let's keep it simple until Rainer gives us a better overview of what a finished solution should look like.

**Review by baha-a — APPROVED** [review]

**Review by 1AhmedYasser — APPROVED** [review]
